### ISSUE TITLE
tvOS: the key values which set with `setValueProvider` disappearing while switching between `.automatic` and `.mainThread`

### ISSUE THREAD

**Issue by joohae-kim** [airbnb/lottie-ios#2163]
## Which Version of Lottie are you using?

Lottie 4.2.0

## Expected Behavior 
The values set by `setValueProvider` are kept internally and transferred between Animation Layers

## Actual Behavior
1. The following code is not working on tvOS because Lottie created `CoreAnimationLayer` when assigning `animation` of the view
2. In `CoreAnimationLayer.display()`, it fails to `setupAnimation()` due `The Core Animation rendering engine doesn't currently support applying trims to filled shapes`
3. create `MainThreadAnimationLayer` for fallback handling
4. the `valueProviders` are not transferred to the new animation layer

```Swift
    let savingAnimationView = LottieAnimationView()
    savingAnimationView.animation = LottieAnimation.named("CloudUpdating")
    let colorProvider = ColorValueProvider(UIColor(verkada: .charcoal).lottieColorValue)
    let keyPath = AnimationKeypath(keypath: "**.Color")
    savingAnimationView.setValueProvider(colorProvider, keypath: keyPath)
    savingAnimationView.play()
    savingAnimationView.loopMode = .loop
    addArrangedSubview(savingAnimationView)
    savingAnimationView.size(CGSize(width: 40, height: 40))
```

I have added the code at the end of `fileprivate func makeAnimationLayer(usingEngine renderingEngine: RenderingEngineOption)` method of `LottieAnimationView`, but not sure it will not cause any issue

```Swift
    valueProviders.forEach({ valueProvider in
      animationLayer.setValueProvider(valueProvider.value, keypath: valueProvider.key)
    })
```

**Comment by calda** [comment]
Thanks for the report. Fixed in #2164.

**Comment by joohae-kim** [comment]
thank you it works on my code also
